src/UserSettingsPage.h

```diff
@@ -353,89 +354,125 @@ class UserSettings : public QObject
     static QSharedPointer<UserSettings> instance_;
 };
 
-class HorizontalLine : public QFrame
+class UserSettingsModel : public QAbstractListModel
 {
     Q_OBJECT
 
-public:
-    HorizontalLine(QWidget *parent = nullptr);
-};
-
-class UserSettingsPage : public QWidget
-{
-    Q_OBJECT
+    enum Indices
+    {
+        GeneralSection,
+        Theme,
+        MobileMode,
+#ifndef Q_OS_MAC
+        ScaleFactor,
+#endif
+        Font,
+        FontSize,
+        EmojiFont,
+        AvatarCircles,
+        UseIdenticon,
+        PrivacyScreen,
+        PrivacyScreenTimeout,
+
+        TimelineSection,
+        TimelineMaxWidth,
+        MessageHoverHighlight,
+        EnlargeEmojiOnlyMessages,
+        AnimateImagesOnHover,
+        TypingNotifications,
+        ReadReceipts,
+        ButtonsInTimeline,
+        Markdown,
+
+        SidebarSection,
+        GroupView,
+        SortByImportance,
+        DecryptSidebar,
+
+        TraySection,
+        Tray,
+        StartInTray,
+
+        NotificationsSection,
+        DesktopNotifications,
+        AlertOnNotification,
+
+        VoipSection,
+        UseStunServer,
+        Microphone,
+        Camera,
+        CameraResolution,
+        CameraFrameRate,
+        Ringtone,
+
+        EncryptionSection,
+        OnlyShareKeysWithVerifiedUsers,
+        ShareKeysWithTrustedUsers,
+        SessionKeys,
+        UseOnlineKeyBackup,
+        OnlineBackupKey,
+        SelfSigningKey,
+        UserSigningKey,
+        MasterKey,
+        CrossSigningSecrets,
+        DeviceId,
+        DeviceFingerprint,
+
+        LoginInfoSection,
+        UserId,
+        Homeserver,
+        Profile,
+        Version,
+        Platform,
+        COUNT,
+        // hidden for now
+        AccessToken,
+#ifdef Q_OS_MAC
+        ScaleFactor,
```

Comment:
Is ScaleFactor meant to be in here twice?

Reply:
Yes, one is only there on MacOS and one on every other platform. That basically hides that setting without having to do any code changes in the data and setData functions.